### PR TITLE
[App Service] `az webapp config backup delete`: Add new command to delete a backup of the webapp

### PR DESCRIPTION
**Pull request by vkarasala**
**Related command**

`webapp config backup delete` is the new command introduced with this PR

**Description**

The new command enables customers to delete a backup by id

**Testing Guide**

List the web app backups by using the command `az webapp config backup list`
Sample commands :
For the app : `az webapp config backup list -g test_group --webapp-name test-container-appa`
For the app's slot : `az webapp config backup list -g test_group --webapp-name test-container-appa -s dev`

Pick an an backup from the list and delete the backup using the command `az webapp config backup delete`
For the app : `az webapp config backup delete -g test_group --webapp-name test-container-appa --backup-id 89215`
For the app's slot : `az webapp config backup delete -g test_group --webapp-name test-container-appa --backup-id 89216 --slot dev`

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Comment by sulabh-msft** [comment]
@vkarasala Can you please check if you can add the tests ?
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_backup_commands.py

**Review by sulabh-msft — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -1161,6 +1161,11 @@
 short-summary: Restore a web app from a backup.
 """
 
+helps['webapp config backup delete'] = """
+type: command
+short-summary: Deletes a web app backup.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1166]: Please use the first-person voice in the help message
```suggestion
short-summary: Delete a web app backup.
```

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 2]
```diff
@@ -197,6 +197,7 @@ def load_command_table(self, _):
         g.custom_command('create', 'create_backup', exception_handler=ex_handler_factory())
         g.custom_command('update', 'update_backup_schedule', exception_handler=ex_handler_factory())
         g.custom_command('restore', 'restore_backup', exception_handler=ex_handler_factory())
+        g.custom_command('delete', 'delete_backup', exception_handler=ex_handler_factory())
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:200]: Do we need to consider adding a `confirmation=True` for the delete command?

**Comment by zhoxing-ms** [comment]
@vkarasala Please note that since we will launch the release of this sprint the day after tomorrow. If you cannot address all the comments tomorrow, we will have to postpone the release of this PR to the next sprint **(03-07)**

**Comment by vkarasala** [comment]
> @vkarasala Please note that since we will launch the release of this sprint the day after tomorrow. If you cannot address all the comments tomorrow, we will have to postpone the release of this PR to the next sprint **(03-07)**

@zhoxing-ms Thanks for the review, resolved your comments.

**Comment by zhoxing-ms** [comment]
> Can you please check if you can add the tests ?
https://github.com/Azure/azure-cli/tree/dev/src/azure-cli/azure/cli/command_modules/appservice/tests/latest

@vkarasala Could you please address this comment?

**Comment by zhoxing-ms** [comment]
@vkarasala Could you resolve the CI issues?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**